### PR TITLE
File printer

### PR DESCRIPTION
**Pull request by Frickerson**

### CODE REVIEW

File: Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#include "Core/CoreExtFilePrinterObject.h"
+
+#include "HAL/PlatformFileManager.h"
```
Comment (TheEmidee) [inline Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp]: <>

File: Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+#include "Misc/FileHelper.h"
```
Comment (TheEmidee) [inline Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp]: <>

File: Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+UENUM( BlueprintType )
+enum ECoreExtCreateFilePolicy
+{
+    Append,
+    Clear,
+    CreateNew
+};
+
+UCLASS( BlueprintType )
+class COREEXTENSIONS_API UCoreExtFilePrinterObject : public UObject
```
Comment (TheEmidee) [inline Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h]: final

File: Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+{
+    GENERATED_BODY()
+
+public:
+    UCoreExtFilePrinterObject();
+
+    UFUNCTION( BlueprintCallable, meta = ( ReturnDisplayName = "Success" ) )
+    static bool CreateFile( UCoreExtFilePrinterObject *& file_printer, FString file_name, const FString sub_directory, ECoreExtCreateFilePolicy creation_policy );
```
Comment (TheEmidee) [inline Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h]: all the strings should be passed by const ref in the function parameters

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp (hunk 1 above)
Comment (Frickerson) [reply Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp]: Done.

File: Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp (hunk 2 above)
Comment (Frickerson) [reply Source/CoreExtensions/Private/Core/CoreExtFilePrinterObject.cpp]: Done.

File: Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h (hunk 3 above)
Comment (Frickerson) [reply Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h]: Done.

File: Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h (hunk 4 above)
Comment (Frickerson) [reply Source/CoreExtensions/Public/Core/CoreExtFilePrinterObject.h]: Done.

**Review by TheEmidee — APPROVED** [review]

**Pull request merged**